cmd/clusterctl/Dockerfile

```diff
@@ -11,22 +11,26 @@
 # limitations under the License.
 
 # Build the cluster binary
-FROM golang:1.10.3 as builder
+FROM golang:1.12.5 as builder
 
 # Copy in the go src
 WORKDIR /go/src/sigs.k8s.io/cluster-api-provider-ibmcloud
 COPY pkg/    pkg/
 COPY cmd/    cmd/
 COPY vendor/ vendor/
 
-# Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o clusterctl sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl
+# install kind and build clusterctl
+RUN GO111MODULE="on" go get sigs.k8s.io/kind@v0.3.0 \
+    && CGO_ENABLED=0 GOOS=linux go build -a -o clusterctl sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl
 
 # Copy clusterctl into a thin image
 FROM alpine:latest
-RUN apk update && apk add ca-certificates openssh-client && rm -rf /var/cache/apk/*
 
-WORKDIR /root/
+RUN apk add --no-cache ca-certificates openssh-client curl \
+    && curl -L https://download.docker.com/linux/static/stable/x86_64/docker-17.09.1-ce.tgz | tar --strip-components=1 -xvz -C /bin/ docker/docker \
+    && curl -L -o /bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.14.2/bin/linux/amd64/kubectl \
```

Comment:
we can download the kind here as well, and the command like this:
`curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/v0.3.0/kind-arch-xxxxx`